contracts/MarginAccount.sol

```diff
@@ -0,0 +1,146 @@
+pragma solidity =0.6.10;
+
+import {SafeMath} from "./packages/oz/SafeMath.sol";
+
+/**
+ * SPDX-License-Identifier: UNLICENSED
+ * @dev The MarginAccount is a library that provides Controller with an Account of Vault structs, and
+ * the functions that manipulate vaults. Vaults describe positions that users have.
+ */
+library MarginAccount {
+    using SafeMath for uint256;
+
+    struct Account {
+        address owner;
+        uint256 vaultIds;
+    }
+
+    struct Vault {
+        uint256[] shortAmounts;
+        uint256[] longAmounts;
+        uint256[] collateralAmounts;
+        address[] shortOtokens;
+        address[] longOtokens;
+        address[] collateralAssets;
+    }
+
+    /**
+     * @dev Increment vault numbers in account
+     */
+    function openNewVault(Account storage _account) internal {
+        _account.vaultIds.add(1);
+    }
+
+    /**
+     * @dev increase the short oToken balance in a vault when a new oToken is minted.
+     */
+    function addShort(
+        Vault storage _vault,
+        address shortOtoken,
+        uint256 amount,
+        uint256 index
+    ) internal {
+        //Check the adding token is the same as vault.shortOtokens[idx] if it exists. (isn't that the same as the below?)
+        //Reverts if vault.shortOtokens[index] != shortOtoken && vault.shortOtoken[index] !== address(0)
+        require(_vault.shortOtokens[index] == shortOtoken);
+        require(_vault.shortOtokens[index] != address(0));
```

Comment:
This one will always revert if a user is depositing more short of an already existent one in the vault.